pkg/virt-controller/watch/vm/vm.go

```diff
@@ -3254,3 +3260,184 @@ func (c *Controller) handleMemoryHotplugRequest(vm *virtv1.VirtualMachine, vmi *
 
 	return nil
 }
+
+func hasEphemeralHotplugVolumes(vm *virtv1.VirtualMachine, vmi *virtv1.VirtualMachineInstance) bool {
+	// hotplug update strategy can handle this case
+	if vm.Spec.UpdateVolumesStrategy != nil && *vm.Spec.UpdateVolumesStrategy == virtv1.UpdateVolumesStrategyHotplug {
+		return false
+	}
+
+	hotplugOp := false
+	volsVM := storagetypes.GetVolumesByName(&vm.Spec.Template.Spec)
+
+	for _, volume := range vmi.Spec.Volumes {
+		hotpluggableVol := (volume.VolumeSource.PersistentVolumeClaim != nil &&
+			volume.VolumeSource.PersistentVolumeClaim.Hotpluggable) ||
+			(volume.VolumeSource.DataVolume != nil && volume.VolumeSource.DataVolume.Hotpluggable)
+		_, ok := volsVM[volume.Name]
+		if !ok && hotpluggableVol {
+			hotplugOp = true
```

Comment:
Can't you just return true here? and skip the `hotplugOp` variable entirely. And return false at the end?

Reply:
Sure, I just copied/pasted that preexisting code into a separate function, did not put any thought into optimizing it

Reply:
done